src/R2_alg.jl

```diff
@@ -162,7 +277,7 @@ function R2(
 
     # define model
     φk(d) = dot(∇fk, d)
-    mk(d) = φk(d) + ψ(d)
+    mk(d)::R = φk(d) + ψ(d)::R
 
     prox!(s, ψ, mν∇fk, ν)
```

Comment:
Some allocations here as well, maybe also related to ProximalOperators.

Reply:
Probably ShiftedProximalOperators?!

Reply:
https://github.com/JuliaSmoothOptimizers/ShiftedProximalOperators.jl/pull/110

Reply:
Are the allocations gone now?

Reply:
There are still a few (16 per iteration on line 280  and 282).
The biggest number of allocations comes from `ψ = shifted(h, xk, l_bound_m_x, u_bound_m_x, selected)` at the begging of R2, but it is not inside a loop. We should think about a way of preallocating the ShiftedProximalOperators.